src/models/v2/security-requirement.ts

```diff
@@ -1,15 +1,14 @@
 import { BaseModel } from '../base';
+
 import type { SecuritySchemeInterface } from '../security-scheme';
-import type { v2 } from "../../spec-types";
-import { SecurityRequirementInterface } from 'models/security-requirement';
-import { SecurityScheme } from './security-scheme';
+import type { SecurityRequirementInterface } from '../security-requirement';
 
-export class SecurityRequirement extends BaseModel<{}, { id: string, scheme: SecuritySchemeInterface }> implements SecurityRequirementInterface {
+export class SecurityRequirement extends BaseModel<{ scopes?: string[], scheme: SecuritySchemeInterface }, { id?: string }> implements SecurityRequirementInterface {
```

Comment:
Good point, so we declare the model directly here so the `_json` can contain the whole object. 👍 right?

Reply:
Yep, you can pass `scopes` and `scheme` directly to the constructor `_json`, not to the metadata.

Reply:
Nice, good improvement 👍